core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3549,7 +3549,17 @@ protected final void createAggImpl(
     // implement HAVING (we have already checked that it is non-trivial)
     relBuilder.push(bb.root());
     if (havingExpr != null) {
-      relBuilder.filter(havingExpr);
+      /* Set the correlation variables used in this sub-query to the filter node,
+       * same logic is being used for the filter generated in where clause.
+       */
+      Set<CorrelationId> variableSet = new HashSet<>();
```

Comment:
Would it possible to have some new test(s) in `SqlToRelConverterTest.java` showing the problem that this piece of code fixes?

Reply:
Without this fix there is an existing test case failure (I think it is in CoreQuidemTest).

Reply:
Yes, correct, `misc.iq` would fail without this code, as described on the other conversation. Thanks for the clarification.